aten/src/ATen/core/boxing/impl/test_helpers.h

```diff
@@ -24,11 +24,18 @@ inline at::Tensor dummyTensor(c10::DispatchKeySet ks) {
       allocator->allocate(size_bytes),
       allocator,
       /*resizable=*/true);
-  return at::detail::make_tensor<c10::TensorImpl>(storage_impl, ks, dtype);
+  at::Tensor t = at::detail::make_tensor<c10::TensorImpl>(storage_impl, ks, dtype);
+  // TODO: We currently have this hack since by default all tensors have a Autograd backend key
+  //       added in TensorImpl constructor. We should remove this hack once we move to only have
+  //       Autograd backend key when the tensor requires_grad=true.
```

Comment:
Hmm. I'm guessing you added this to make some of the tests work out, but it is kind of confusing to have the tests exercising a feature (tensors without the autograd key on them) which don't reflect how the actual system works. I wouldn't even call it a hack... more like, these tests come from the future :>